### PR TITLE
Update RMS normalization and reduce_mean tests

### PR DESCRIPTION
**Pull request by chandrasekaranpradeep**
1.  #125 - Test passed(i.e [forge/test/mlir/llama/tests/test_llama_rotary_emb.py](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/llama/tests/test_llama_rotary_emb.py))

2. #114
        There are two test cases in the [forge/test/mlir/test_ops.py](https://github.com/tenstorrent/tt-forge-fe/blob/main/forge/test/mlir/test_ops.py)
        [test_reduce_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L361)
In this test cases, **TTNN: Issues with ttnn implicit reshape is resolved**
The below test cases alone is failing with pcc issue (PCC: 0.7203957195745748)
`pytest forge/test/mlir/test_ops.py::test_reduce_mean[-1-input_shape2] -vss `
[test_mean](https://github.com/tenstorrent/tt-forge-fe/blob/3817b9b2e559cb97497e035c7cf62e69bac455b3/forge/test/mlir/test_ops.py#L416)
In this test cases, the **FFE: Unsupported squeeze operation** issue is resolved, now all the test cases here are failing with `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first` issue.

### CODE REVIEW

File: forge/test/mlir/test_ops.py [hunk 1]
```diff
@@ -409,16 +409,11 @@ def forward(self, x, y):
...
-
+@pytest.mark.xfail(reason="Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first")
 @pytest.mark.parametrize("x_shape", [7, 32, 41])
 @pytest.mark.parametrize("y_shape", [7, 32, 41])
 @pytest.mark.parametrize("dim", [1, 2])
 def test_mean(x_shape, y_shape, dim):
-    if dim == 1:
-        pytest.skip("FFE: Unsupported squeeze operation")
-    if dim == 2:
-        # Note: Some tests are passing when run in group, while failing when running individually
```
Comment (chandrasekaranpradeep) [inline forge/test/mlir/test_ops.py:420]: @nvukobratTT This test are failing with `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first` issue now.
At dim = -2, earlier it was failed with  TTNN: Tensor layout bugs. Are they referring to above mentioned issue by myself earlier? I can't find issue/blocker attached to exact layout issue.

File: forge/test/mlir/test_ops.py [hunk 2]
```diff
@@ -409,16 +409,11 @@ def forward(self, x, y):
...
-        pytest.skip("TTNN: Tensor layout bugs")
```
Comment (chandrasekaranpradeep) [inline forge/test/mlir/test_ops.py:421]: @nvukobratTT This test are failing with `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first` issue now.
At dim = -2, earlier it was failed with  TTNN: Tensor layout bugs. Are they referring to above mentioned issue by myself earlier? I can't find issue/blocker attached to exact layout issue.

File: forge/test/mlir/test_ops.py [hunk 3]
```diff
@@ -359,10 +359,10 @@ def forward(self, a):
     -2,
 ])
 def test_reduce_mean(input_shape, dim):
-    # Check if certain any dim in input_shape isn't divisible by 32
-    if any(i % 32 != 0 for i in input_shape[1:]): 
-        pytest.xfail("TTNN: Issues with ttnn implicit reshape") 
-    
+
+    if input_shape == (1, 12, 3200) and dim == -1:
+        pytest.xfail("Data mismatch between PyTorch and TTNN (PCC: 0.72)")
```
Comment (nvukobratTT) [inline forge/test/mlir/test_ops.py:364]: Can you create issues on TT-MLIR and link them in this XFail command? Feel free to do the same for another xfail for the op test. 

Also, update blocker field of reduce_avg op issues with these two as blockers :)) Throught TT-MLIR issues, we'll see if someone from that team can debug further. If not, we're going to push them further as well :))

**Review by nvukobratTT — APPROVED** [review]

File: forge/test/mlir/test_ops.py (hunk 3 above)
Comment (chandrasekaranpradeep) [reply forge/test/mlir/test_ops.py:364]: Sure Nikola. I will create a issue in TT-MLIR and add them in XFail and blocker field of the reduce_avg Op issue.

Comment (chandrasekaranpradeep) [reply forge/test/mlir/test_ops.py:364]: @nvukobratTT 
1. Raised issue for data mismatch in TT-MLIR repo and updated blocker field of reduce_avg op- https://github.com/tenstorrent/tt-mlir/issues/869.
2. For `Unable to reshape a tensor in TILE_LAYOUT to non-tile height and width! Please convert the tensor to ROW_MAJOR_LAYOUT first` issue from another mean op test cases will be resolved in latest tt-metal main.
I have checked by creating TTNN test from the TTIR generated for the above layout issue and now I am not seeing any layout issue in the TTNN test cases. But for the same TTNN test cases, earlier I am able to reproduce the same layout issue.
The layout issue is similar to ttnn.embedding op layout issue - https://github.com/tenstorrent/tt-metal/issues/12644
Once the tt-metal latest main is uplifted in tt-mlir repo, I will check the test cases and then create a issue if needed
Is it works for @nvukobratTT ? 

**TTIR generated for the mean layout issue:**
```
module @Mean attributes {tt.system_desc = #tt.system_desc<[{arch = <wormhole_b0>, grid = 8x8, l1_size = 1499136, num_dram_channels = 12, dram_channel_size = 1073741824, noc_l1_address_align_bytes = 16, pcie_address_align_bytes = 32, noc_dram_address_align_bytes = 32, l1_unreserved_base = 1024, erisc_l1_unreserved_base = 1024, dram_unreserved_base = 1024, dram_unreserved_end = 1073741824, physical_cores = {worker = [ 0x0,  0x1,  0x2,  0x3,  0x4,  0x5,  0x6,  0x7,  1x0,  1x1,  1x2,  1x3,  1x4,  1x5,  1x6,  1x7,  2x0,  2x1,  2x2,  2x3,  2x4,  2x5,  2x6,  2x7,  3x0,  3x1,  3x2,  3x3,  3x4,  3x5,  3x6,  3x7,  4x0,  4x1,  4x2,  4x3,  4x4,  4x5,  4x6,  4x7,  5x0,  5x1,  5x2,  5x3,  5x4,  5x5,  5x6,  5x7,  6x0,  6x1,  6x2,  6x3,  6x4,  6x5,  6x6,  6x7,  7x0,  7x1,  7x2,  7x3,  7x4,  7x5,  7x6,  7x7] dram = [ 8x0,  9x0,  10x0,  8x1,  9x1,  10x1,  8x2,  9x2,  10x2,  8x3,  9x3,  10x3]}, supported_data_types = [<f32>, <f16>, <bf16>, <bfp_f8>, <bfp_bf8>, <bfp_f4>, <bfp_bf4>, <bfp_f2>, <bfp_bf2>, <u32>, <u16>, <u8>], supported_tile_sizes = [ 4x16,  16x16,  32x16,  4x32,  16x32,  32x32]}], [0], [3 : i32], [ 0x0x0x0]>} {
  func.func @forward(%arg0: tensor<1x7x7xf32> {ttir.name = "x"}) -> (tensor<1x7xf32> {ttir.name = "Mean.output_squeeze_1"}) {
    %0 = tensor.empty() : tensor<1x1x7xf32>
    %1 = "ttir.mean"(%arg0, %0) <{dim_arg = [-2 : i32], keep_dim = true, operand_constraints = [#tt.operand_constraint<dram|l1|scalar|tile|none|interleaved|single_bank|height_sharded|width_sharded|block_sharded|any_layout|any_device|any_device_tile|l1_block_sharded>, #tt.operand_constraint<dram|l1|scalar|tile|none|interleaved|single_bank|height_sharded|width_sharded|block_sharded|any_layout|any_device|any_device_tile|l1_block_sharded>]}> : (tensor<1x7x7xf32>, tensor<1x1x7xf32>) -> tensor<1x1x7xf32>
    %2 = tensor.empty() : tensor<1x7xf32>
    %3 = "ttir.squeeze"(%1, %2) <{dim = 1 : si32, operand_constraints = [#tt.operand_constraint<dram|l1|scalar|tile|none|interleaved|single_bank|height_sharded|width_sharded|block_sharded|any_layout|any_device|any_device_tile|l1_block_sharded>, #tt.operand_constraint<dram|l1|scalar|tile|none|interleaved|single_bank|height_sharded|width_sharded|block_sharded|any_layout|any_device|any_device_tile|l1_block_sharded>]}> : (tensor<1x1x7xf32>, tensor<1x7xf32>) -> tensor<1x7xf32>
    return %3 : tensor<1x7xf32>
  }
}
```
 
 **TTNN test cases for the above TTIR:**
```
@pytest.mark.parametrize("x_shape", [7, 32, 41])
@pytest.mark.parametrize("y_shape", [7, 32, 41])
@pytest.mark.parametrize("dim", [1, 2])
def test_mean_layout_issue(device, x_shape, y_shape, dim):
    torch.manual_seed(0)
    input_shape = (1, x_shape, y_shape)
    reduce_dim = dim
    
    # CPU
    torch_input_tensor = torch.rand(input_shape, dtype=torch.float32)
    torch_output_tensor = torch.mean(torch_input_tensor, dim=reduce_dim, keepdim=True)
    torch_output_tensor = torch.squeeze(torch_output_tensor, dim=reduce_dim)
    
    # TTNN
    ttnn_input_tensor = ttnn.from_torch(torch_input_tensor, dtype=ttnn.float32, layout=ttnn.TILE_LAYOUT, device=device)
    ttnn_output_tensor = ttnn.mean(ttnn_input_tensor, dim=reduce_dim)
    ttnn_output_tensor = ttnn.squeeze(ttnn_output_tensor, dim=reduce_dim)
    ttnn_output_tensor = ttnn.to_torch(ttnn_output_tensor)

    assert_with_pcc(torch_output_tensor, ttnn_output_tensor, pcc=0.99)
```

Comment (chandrasekaranpradeep) [reply forge/test/mlir/test_ops.py]: Synced offline with @nvukobratTT
